lisp/ess-tracebug.el

```diff
@@ -1264,8 +1269,14 @@ If in debugging state, mirrors the output into *ess.dbg* buffer."
       (setq string (replace-regexp-in-string prompt-replace-regexp " \n" string nil nil 1))
 
       ;; replace long prompts
-      (when inferior-ess-replace-long+
-        (setq string (replace-regexp-in-string "\\(\\+ \\)\\{4\\}\\(\\+ \\)+" ess-long+replacement string)))
+      (cond ((and (eq inferior-ess-replace-long+ 'strip)
+                  (or (not ess-eval-visibly)
+                      (eq ess-eval-visibly 'nowait)))
+             (setq string (replace-regexp-in-string "\\(\\+[[:space:]]\\).*\\'"
+                                                    "" string)))
```

Comment:
Won't work either ;).  `+ + +` need not occur at the end of the process filter `string`. I think the best you can do here is to match `+ ` at least twice. 

Also could you please use a plain space instead of `[[:space:]]`. We know for sure that it's a space there, it's more readable and it's likely to be slightly more efficient.

Reply:
The newest version (just pushed) seems to work for me, though it does occasionally put the output on the line containing the prompt instead of the line after that. Can you give an example of where it fails?

Reply:
For example if the string is `+ + + + > some-output >`  you will strip all of the output. It will also strip prompt in usual `+ + + >` in which case there will be no indication of evaluation in the REPL. Some emacs ides do that but I don't like it personally.   

I would rather use `\\(\\+ \\)\\{2,\\}` there. In any case you will still have the "offset" problem. 

The solution is not as simple as it looks. I don't think it could be solved with reg-exp in process filter only. One needs to look at what follows after + + + and insert an "\n". It also should be efficient. The best way I can think of is to search for + + + just before the accumulated output is inserted into the repl (see accumulation code nearby) .

Reply:
To clarify it further, the regexp handles the limiting case when "+ + >" arrives in one chunk and the output in another. That is probably happening in your tests but doesn't hold in general. 

I will pull your branch over the weekend and integrate the fix on the accumulated output which I described above. 

Besides the mtcars example do you have other examples with shifted output which I can try out?